src/backend/distributed/executor/multi_utility.c

```diff
@@ -3819,6 +3819,8 @@ ProcessDropTableStmt(DropStmt *dropTableStatement)
 			continue;
 		}
 
+		EnsureCoordinator();
```

Comment:
Actually we need to ensure that partitioned table has no partitions. Because we send `DROP TABLE partitioned_table` commands to workers and that DROP command is also follows this path. If we put `EnsureCoordinator()` before `partitionList = PartitionList(relationId);`, we will error out for those.